### PR TITLE
Added `--conservative-mode` config option

### PR DESCRIPTION
**Pull request by patocallaghan**
Fixes https://github.com/ember-codemods/ember-no-implicit-this-codemod/issues/11

Introduces a `dont-assume-this` config option which does not attempt to transform a `PathExpression` with `this.` if its name does not appear in a component's telemetry data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,12 @@ yarn global add ember-no-implicit-this-codemod
 ember-no-implicit-this-codemod <TRANSFORM NAME> path/of/files/ or/some**/*glob.js
 ```
 
+The codemod accepts the following options:
+
+| Option | Value | Default | Details |
+| --- | --- | ---| --- |
+| --conservative-mode | boolean | false | The codemod won't prepend `this.` to `PathExpressions` names which don't appear in a component's telemetry data |
```
Comment (patocallaghan) [inline README.md]: I'm definitely not tied to `conservative-mode` so please let me know if you think there's a better name. Naming is hard 😅

Comment (NullVoxPopuli) [reply README.md]: maybe `--dont-assume-this-if-property-not-found`?

Comment (patocallaghan) [reply README.md]: That seems a bit verbose and requires _lots_ of typing 😬

How about `--dont-assume-this` instead? Hopefully the description of the option in the table is descriptive enough that we don't need to fully describe it in the option name? I can update the description if not.

On the other hand, if you feel strongly about using `--dont-assume-this-if-property-not-found` I'll make the change.

**Comment by NullVoxPopuli** [comment]
I think don't assume this is good :+1:

**Comment by patocallaghan** [comment]
@NullVoxPopuli Thanks I've updated the option to be `--dont-assume-this`

**Comment by NullVoxPopuli** [comment]
Thanks for working on this!

**Pull request merged**